source/common/upstream/upstream_impl.h

```diff
@@ -60,36 +61,25 @@ class HealthCheckHostMonitorNullImpl : public HealthCheckHostMonitor {
 /**
  * Implementation of Upstream::HostDescription.
  */
-class HostDescriptionImpl : virtual public HostDescription {
+class HostDescriptionImpl : virtual public HostDescription,
+                            protected Logger::Loggable<Logger::Id::upstream> {
 public:
   HostDescriptionImpl(
       ClusterInfoConstSharedPtr cluster, const std::string& hostname,
       Network::Address::InstanceConstSharedPtr dest_address,
       const envoy::api::v2::core::Metadata& metadata,
       const envoy::api::v2::core::Locality& locality,
       const envoy::api::v2::endpoint::Endpoint::HealthCheckConfig& health_check_config,
-      uint32_t priority)
-      : cluster_(cluster), hostname_(hostname), address_(dest_address),
-        canary_(Config::Metadata::metadataValue(metadata, Config::MetadataFilters::get().ENVOY_LB,
-                                                Config::MetadataEnvoyLbKeys::get().CANARY)
-                    .bool_value()),
-        metadata_(std::make_shared<envoy::api::v2::core::Metadata>(metadata)), locality_(locality),
-        locality_zone_stat_name_(locality.zone(), cluster->statsScope().symbolTable()),
-        priority_(priority) {
-    if (health_check_config.port_value() != 0 &&
-        dest_address->type() != Network::Address::Type::Ip) {
-      // Setting the health check port to non-0 only works for IP-type addresses. Setting the port
-      // for a pipe address is a misconfiguration. Throw an exception.
-      throw EnvoyException(
-          fmt::format("Invalid host configuration: non-zero port for non-IP address"));
-    }
-    health_check_address_ =
-        health_check_config.port_value() == 0
-            ? dest_address
-            : Network::Utility::getAddressWithPort(*dest_address, health_check_config.port_value());
+      uint32_t priority);
+
+  Network::TransportSocketFactory&
```

Comment:
move this to private? seems not used outside

Reply:
done.